boilerplate/app/components/screen/screen.tsx

```diff
@@ -32,6 +32,37 @@ function ScreenWithScrolling(props: ScreenProps) {
   const backgroundStyle = props.backgroundColor ? { backgroundColor: props.backgroundColor } : {}
   const insetStyle = { paddingTop: props.unsafe ? 0 : insets.top }
 
+  // The followings for <Screen preset='auto'/>
+  // This will automatically disables scrolling if content fits the screen.
+  const { height } = Dimensions.get('window')
+  const screenHeight = React.useRef(null)
+  const [scrollEnabled, setScrollEnabled] = React.useState(true)
+
+  const updateScrollState = () => {
+    if (props.preset === 'auto'){
+      // check whether if content fits the screen
+      const contentFitsScreen = screenHeight.current > height * presets.auto.offset.percent - presets.auto.offset.point
+        
+      // then toggle scroll state according to it, make sure it's not rendering twice
+      if (scrollEnabled && !contentFitsScreen) setScrollEnabled(false)
+      else if (!scrollEnabled && contentFitsScreen) setScrollEnabled(true)
+    }
+    else if (!scrollEnabled) setScrollEnabled(true) // set back initial state in case it's locked 
```

Comment:
Nit: I prefer either one-liners or using braces, rather than 2-liners.

```suggestion
    } else if (!scrollEnabled) {
      // set back initial state in case it's locked
      setScrollEnabled(true)
    }
```